ArcGISRuntimeSDKQt_CppSamples/UtilityNetwork/ValidateUtilityNetworkTopology/ValidateUtilityNetworkTopology.cpp

```diff
@@ -0,0 +1,669 @@
+}
+
+void ValidateUtilityNetworkTopology::onServiceGeodatabaseLoaded(ServiceVersionParameters* params, const Error& loadError)
+{
+  if (!loadError.isEmpty())
+    return;
+
+  m_serviceGeodatabase->createVersionAsync(params).then(this, [this](ServiceVersionInfo* serviceVersionInfo)
+  {
+    m_serviceGeodatabase->switchVersionAsync(serviceVersionInfo->name()).then(this, [this]()
+    {
+      displayLabelDefinitions();
+
+      // Visualize dirty area by adding to the map
+      ArcGISFeatureTable* m_dirtyAreaTable = m_utilityNetwork->dirtyAreaTable();
+      m_map->operationalLayers()->append(new FeatureLayer(dynamic_cast<FeatureTable*>(m_dirtyAreaTable), this));
+    });
+  });
+}
+
+void ValidateUtilityNetworkTopology::connectSignals()
+{
+  connect(m_mapView, &MapQuickView::mouseClicked, this, [this](QMouseEvent& mouseEvent)
+  {
+    // first clear the selection
+    m_deviceFeatureLayer->clearSelection();
+    m_lineFeatureLayer->clearSelection();
+
+    m_busy = true;
+    emit isBusy();
+
+    updateMessage("Identifying feature to edit...");
+
+    if (m_map->loadStatus() != LoadStatus::Loaded)
+      return;
+
+    Point m_clickPoint = m_mapView->screenToLocation(mouseEvent.position().x(), mouseEvent.position().y());
+
+    // Perform an identify to determine if a user tapped on a feature.
+    m_mapView->identifyLayersAsync(mouseEvent.position(), 5.0, false).then(this, [this](const QList<IdentifyLayerResult*>& results)
+    {
+      onIdentifyLayersAsyncCompleted(results);
+    });
+  });
+}
+
+void ValidateUtilityNetworkTopology::onIdentifyLayersAsyncCompleted(const QList<IdentifyLayerResult*>& results)
+{
+  if (!results.isEmpty())
+  {
+    for (IdentifyLayerResult* result : results)
+    {
+      if (result && (result->layerContent()->name() == "Electric Distribution Device" || result->layerContent()->name() == "Electric Distribution Line"))
+      {
+        m_feature = static_cast<ArcGISFeature*>(qAsConst(result)->geoElements().first());
+        break;
+      }
+    }
+  }
+  else
+  {
+    updateMessage("No feature identified. Tap on a feature to edit.");
+
+    m_busy = false;
+    emit isBusy();
+
+    return;
+  }
+  m_updateFieldName = m_feature->featureTable()->tableName() == "Electric Distribution Device" ? "devicestatus" : "nominalvoltage";
+  Field m_field = m_feature->featureTable()->field(m_updateFieldName);
+  CodedValueDomain m_codedValueDomain = static_cast<CodedValueDomain>(m_field.domain());
+  m_codedValues = m_codedValueDomain.codedValues();
+  if (m_field.isEmpty() || m_codedValues.empty())
+  {
+    return;
+  }
+
+  if (m_feature->loadStatus() != LoadStatus::Loaded)
+    m_feature->load();
+
+  m_featureToEdit = m_feature;
+
+  // Select the feature.
+  if (m_updateFieldName == "nominalvoltage")
+    m_lineFeatureLayer->selectFeature(m_featureToEdit);
+  else if (m_updateFieldName == "devicestatus")
+    m_deviceFeatureLayer->selectFeature(m_featureToEdit);
+
+  m_choices.clear();
+
+  m_choices.append("");
```

Comment:
why do you need to append an empty string?

Reply:
Here, to get the attributevalue of the selected feature as the current option in the choices dropdown, first I'm assigning a empty value and replacing it with the feature's attribute value when it's found in the for loop